server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -298,6 +299,8 @@
      */
     public SearchModule(Settings settings, List<SearchPlugin> plugins) {
         this.settings = settings;
+        //TODO: constructor call goes here
+        this.valuesSourceRegistry = new ValuesSourceRegistry();
```

Comment:
Do you still like the idea of changing the ctor to take an `Iterable`? Is that a thing for a follow up change?

Reply:
I didn't see a clean way to do this.  We already manually list all the aggregations once in `registerAggregations`, and that still feels like the right place to hook in the values source mappings for each aggregation.  To do that in the constructor, I'd basically need to duplicate that list.  I thought briefly about some kind of visitor pattern to cover both uses, but it seemed like a lot of boiler plate that didn't add any clarity.  If you've got a good idea for how we could make this cleaner, I'm happy to work with you on it, but I don't see a better way than what we're doing now.

Reply:
I think it is cool to merge as is and grab that in a follow up. Would it be ok if I gave it a shot myself once this is in?

Reply:
Absolutely!